libqtile/core/manager.py

```diff
@@ -530,19 +530,54 @@ def current_window(self):
     def add_strut(self, strut):
         from libqtile.bar import Bar, Gap
 
+        x_screen_dimensions = self.root.get_geometry()
+
+        if len(strut) == 12:
+            if strut[0]:  # left
+                x = abs(strut[0])
+                y = abs((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
+            elif strut[1]:  # right
+                x = x_screen_dimensions.width - abs(strut[1])
+                y = abs((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
+            elif strut[2]:  # top
+                x = abs((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
+                y = abs(strut[2])
+            else:  # bottom
+                x = abs((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
+                y = x_screen_dimensions.height - abs(strut[3])
+            screen = self.find_screen(x, y)
+        else:
+            logger.warning("Not enough information to calculate the relevant screen. Defaulting to current.")
+            screen = self.current_screen
+
+        if screen is None or screen.x is None or screen.y is None or \
```

Comment:
Looks like a job for `not all()` :)

Reply:
Python really has some neat features. Going with `any()` even allows to avoid a `not not ` construction.